gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveValidationQueryGenerator.java

```diff
@@ -86,4 +88,81 @@
 
     return queries;
   }
+
+  /***
+   * Generates Hive SQL that can be used to validate the quality between two {@link Table}s or optionally
+   * {@link Partition}. The query returned is a basic join query that returns the number of records matched
+   * between the two {@link Table}s.
+   * The responsibility of actually comparing this value with the expected module should be implemented by
+   * the user.
+   *
+   * @param sourceTable Source Hive {@link Table} name.
+   * @param sourceDb Source Hive database name.
+   * @param targetTable Target Hive {@link Table} name.
+   * @param optionalPartition Optional {@link Partition} to limit the comparison.
+   * @return Query to find number of rows common between two tables.
+   */
+  public static String generateDataValidationQuery(String sourceTable, String sourceDb, Table targetTable,
+      Optional<Partition> optionalPartition) {
+
+    StringBuilder sb = new StringBuilder();
+
+    // Query head
+    sb.append("SELECT count(*) FROM `")
+        .append(sourceDb).append("`.`").append(sourceTable).append("` s JOIN `")
+        .append(targetTable.getDbName()).append("`.`").append(targetTable.getTableName()).append("` t ON \n");
+
+    // Columns equality
+    boolean isFirst = true;
+    List<FieldSchema> fieldList = targetTable.getSd().getCols();
+    for (FieldSchema field : fieldList) {
+
+      // Do not add maps in the join clause // Add open hive ticket here.
+      if (StringUtils.startsWithIgnoreCase(field.getType(), "map")) {
+        continue;
+      }
+
+      if (isFirst) {
+        isFirst = false;
+      } else {
+        sb.append(" AND \n");
+      }
+
+      // Do not add maps in the join clause // Add open hive ticket here.
+      if (StringUtils.startsWithIgnoreCase(field.getType(), "map")) {
+        continue;
+      }
+      String colName = field.getComment().replaceAll("from flatten_source ", "").trim();
```

Comment:
The source column lineage information is available in field's comment. Remove the description prefix "from flatten_source"
Added as comment.